### ISSUE TITLE
LinearGaussianStateSpaceModel sample doesn't work with num_timesteps=1

### ISSUE THREAD

**Issue by jeffpollock9** [tensorflow/probability#417]
Should `tfd.LinearGaussianStateSpaceModel` support `num_timesteps=1`? Currently:

```python
import tensorflow_probability as tfp
import tensorflow as tf


tfd = tfp.distributions
tfl = tf.linalg


num_dims = 3
num_timesteps = 1
step_std = 1.0
noise_std = 5.0

ssm = tfd.LinearGaussianStateSpaceModel(
    num_timesteps=num_timesteps,
    transition_matrix=tfl.LinearOperatorIdentity(num_dims),
    transition_noise=tfd.MultivariateNormalDiag(
        scale_diag=tf.fill([num_dims], tf.square(step_std))
    ),
    observation_matrix=tfl.LinearOperatorIdentity(num_dims),
    observation_noise=tfd.MultivariateNormalDiag(
        scale_diag=tf.fill([num_dims], tf.square(noise_std))
    ),
    initial_state_prior=tfd.MultivariateNormalDiag(scale_diag=tf.ones([num_dims])),
)

sample = ssm.sample()
```

outputs:

```python
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/jeff/workspace/misc/num_timesteps1_bug.py", line 27, in <module>
    sample = ssm.sample()
  File "/home/jeff/.virtualenvs/prematch-models/lib/python3.6/site-packages/tensorflow_probability/python/distributions/distribution.py", line 840, in sample
    return self._call_sample_n(sample_shape, seed, name, **kwargs)
  File "/home/jeff/.virtualenvs/prematch-models/lib/python3.6/site-packages/tensorflow_probability/python/distributions/distribution.py", line 818, in _call_sample_n
    samples = self._sample_n(n, seed, **kwargs)
  File "/home/jeff/.virtualenvs/prematch-models/lib/python3.6/site-packages/tensorflow_probability/python/distributions/linear_gaussian_ssm.py", line 588, in _sample_n
    _, observation_samples = self._joint_sample_n(n, seed=seed)
  File "/home/jeff/.virtualenvs/prematch-models/lib/python3.6/site-packages/tensorflow_probability/python/distributions/linear_gaussian_ssm.py", line 651, in _joint_sample_n
    latents], axis=0)
  File "/home/jeff/.virtualenvs/prematch-models/lib/python3.6/site-packages/tensorflow/python/util/dispatch.py", line 180, in wrapper
    return target(*args, **kwargs)
  File "/home/jeff/.virtualenvs/prematch-models/lib/python3.6/site-packages/tensorflow/python/ops/array_ops.py", line 1271, in concat
    return gen_array_ops.concat_v2(values=values, axis=axis, name=name)
  File "/home/jeff/.virtualenvs/prematch-models/lib/python3.6/site-packages/tensorflow/python/ops/gen_array_ops.py", line 1209, in concat_v2
    _six.raise_from(_core._status_to_exception(e.code, message), None)
  File "<string>", line 3, in raise_from
tensorflow.python.framework.errors_impl.InvalidArgumentError: ConcatOp : Ranks of all input tensors should match: shape[0] = [1,1,3,1] vs. shape[1] = [0] [Op:ConcatV2] name: concat
```

I'm happy to try a fix if this is a genuine bug.

As far as I can tell all other methods seem to work (e.g. `forward_filter`).

Thanks!

**Comment by davmre** [comment]
This is definitely a bug, thanks for reporting it!

A fix would definitely be appreciated. If you get a chance to work on this, we'd humbly request that the fix include a new unit test covering the `num_timesteps=1` case (this can be literally the simple example in your post), so that we catch this bug if it ever recurs in the future.

**Comment by prakashkagitha** [comment]
Hello,

I reproduced the error and looked into the `linear_gaussian_ssm.py` file where the actual bug is. It is because tf.scan in the line below gives empty tensors for latents and observations if `num_timesteps=1` as self.initial_step+1 and self.final_step are equal. (because `self.final_step` = `self.initial_step`+`num_timesteps`)
```
(latents, observations) = tf.scan(
          sample_step,
          elems=tf.range(self.initial_step+1, self.final_step),
          initializer=(initial_latent, initial_observation))
```
So, in lines below concat operation on `latents` doesn't work as the dimensions doesn't match (here [0] for empty tensor). It is straight forward to avoid this. I made those changes in my nightly build and checked with the above example and it worked.

Can I create a pull request with those changes?
